### PR TITLE
Stay in Touch: Milestone 5 

### PR DESCRIPTION
**Pull request by Mihndim2020**
# Description
This project is a collaborative project in which on this milestone we were asked to create a model with associations and all requested features for friendships, [example here](https://smartfunnycool.com/friendships-in-activerecord/).

Therefore this pull request is to determine that we; 
- [x] Define Relationships in the Friendship Model
- [x] Create Has Many Relationships in the User Model
- [x] Create a friends method in the User Model
- [x] Other Helper Methods for Friendships

# How To Enforce standard Guide 
We achieved this with the following:

- [x] Rubocop
- [x] Stylelint
- [x] Test Integration

# Checklist:

- [x] Our code follows the style guidelines of this project
- [x] IWe performed a self-review of our code
- [x] IWe commented on my code, particularly in hard-to-understand areas
- [x] IWe made corresponding changes to the documentation
- [x] IWe added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally 
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mke2111** [comment]
# Project Under review ⚪

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads**. 

### Well-done 💯 
    1. The linter checks pass successfully
    

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] User can invite friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/send_invitation.png))
- [x] User can see friendships requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png))
- [x] User has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png))
- [x] User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png))
- [x] There should a button to invite user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png))
There is currently no button in either views.
- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png))
Currently, your app shows all posts from all users.
- [x] Add more unit and integration tests for your app to the tests that you already have.
- [x] None of your tests is running. This is the error being thrown.
```
Failure/Error: require 'rails_helper'

LoadError:
  cannot load such file -- rails_helper
```

### OPTIONAL

- [x] In the PR message, rubocop and stylelint are not used to test your app but rather enforce standard guidelines in your code.
- [x] You have repeated this statement **I want to be able** 10 times in your project description in the readme file.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: README.md [hunk 1]
```diff
@@ -1,161 +1,161 @@
...
+```
+    rails server
+```
+
+Open `http://localhost:3000/` in your browser.
+
+### Run tests
+
+```
+    rpsec --format documentation
```
Comment (mke2111) [inline README.md]: - [x] Change this command to `rspec --format documentation`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
+      <%= link_to 'Stay in touch', root_path %>
+      <div class="menu">
+        <div class="menu-item">
+          <%= menu_link_to 'Timeline', posts_path %>
+        </div>
+        <div class="menu-item">
+          <%= menu_link_to 'All users', users_path %>
+        </div>
+      </div>
+      <% if current_user %>
```
Comment (mke2111) [inline app/views/layouts/application.html.erb:21]: OPTIONAL

- [ ] It is a good practice to have zero logic in the views directory. Create methods in the helpers directory and call the methods in the views directory instead.
Read this [article](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) for a head-start. Another great piece to read is this [article](https://brandonhilkert.com/blog/refactoring-logic-from-a-rails-view/)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,41 +1,41 @@
...
+        <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
+      <% else %>
+        <%= link_to 'Sign in', user_session_path %>
+      <% end %>
+    </nav>
+    <% if notice.present? %>
```
Comment (mke2111) [inline app/views/layouts/application.html.erb:27]: Another logic statement

File: README.md (hunk 1 above)
Comment (errea) [reply README.md]: okk much appreciated, would fix that

**Comment by Mihndim2020** [comment]
Changes effected as requested. Thank you for your feedback

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @Mihndim2020 and @errea 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes  👇.
- [x] User is not able to send a request, due to which we are not able to check the see, accept, reject the friendship request. Please work on the last code reviewer suggestions.
- [x] There are test cases that are failing. please make sure that all the test cases should be passed.
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad (Code Reviewer)](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by Mihndim2020** [comment]
Changes effected as requested.

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️
Hi @Mihndim2020 and @errea, Weldone for fighting hard to implement what the previous Code Reviewer suggested, Your tests passed. However for this project to be approved this should be working:

- [x] There should be a button to invite user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
kindly implement the suggestion and request for another review

**Comment by Mihndim2020** [comment]
Invite button added as requested. Thank you for your feedback.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
 Hi @Mihndim2020  and @errea 

I can't able to switch to the correct feature branch because of an error I'm getting

```
error: invalid path 'spec/models /friendship_spec.rb'
error: invalid path 'spec/models /user_spec.rb'

```
- [x] After the model there is a space, please remove that. Because of this, I can't able to switch to  feature/ror-social-M5 branch. 


## Please fix that and submit another review request.

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
[Dipesh Kumar](https://deeptwist.tk) | [Twitter](https://twitter.com/97deepeshkumar)

**Comment by Mihndim2020** [comment]
Space removed as requested. Thank you for your feedback.

**Review by rukundoeric — APPROVED** [review]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - Follows gitwork-flow. 💯 

- ### To Improve
  - N/A

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**